dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -965,16 +1012,32 @@ Spdp::data_received(const DataSubmessage& data,
     return;
   }
 
-  handle_participant_data(
-    (data.inlineQos.length() && disposed(data.inlineQos)) ?
-      DCPS::DISPOSE_INSTANCE : DCPS::SAMPLE_DATA,
-    pdata, to_opendds_seqnum(data.writerSN), from, false);
-
+  const bool from_relay = from == config_->spdp_rtps_relay_address();
 #ifdef OPENDDS_SECURITY
+#ifndef OPENDDS_SAFETY_PROFILE in the #ifdef OPENDDS_SECURITY
```

Comment:
I don't know what happened in the last commit but this line isn't right.  Please change it back to what it was before.